OpenRA.Game/Traits/ActivityUtils.cs

```diff
@@ -44,7 +44,7 @@ public static Activity RunActivity(Actor self, Activity act)
 				else
 					start = current;
 
-				if (prev == act)
+				if (act == prev || act == prev.Parent)
```

Comment:
Ok, that makes sense.  You've got a better picture of all the moving parts than me, so I trust that this will work out by the end.  More comments though, please!